src/main/java/jenkins/plugins/logstash/persistence/BuildData.java

```diff
@@ -112,10 +118,17 @@ public TestData(Action action) {
       passCount = totalCount - skipCount - failCount;
 
       failedTests = new ArrayList<String>();
-      failedTestsWithErrorDetail = new ArrayList<FailedTest>();
+      failedTestsWithDetail = new ArrayList<ExecutedTest>();
       for (TestResult result : testResultAction.getFailedTests()) {
           failedTests.add(result.getFullName());
-          failedTestsWithErrorDetail.add(new FailedTest(result.getFullName(),result.getErrorDetails()));
+          failedTestsWithDetail.add(new ExecutedTest(result.getFullName(),result.getErrorDetails(), result.getDuration()));
+      }
+
+      passedTests = new ArrayList<String>();
+      passedTestsWithDetail = new ArrayList<ExecutedTest>();
+      for (TestResult result : testResultAction.getPassedTests()) {
```

Comment:
this isn't dry now, can you please refactor the logic into a helper method that will take in a collection of TestResult and fill a List<String> and List<ExecutedTest>?

Reply:
This is done now, but it throws an unchecked/unsafe warning because I had to cast the return of getPassedTests/getFailedTests to List<TestResult> (the default return seems to be something like "capture #<changing number> - TestResult"). Is there a better way to handle this or is it expected?